spec/mongo/bulk_write_spec.rb

```diff
@@ -348,6 +348,30 @@
           end
 
           context 'when bulk executing update_one' do
+            context 'with hint option' do
+              let(:requests) do
+                [{
+                   update_one: {
+                     filter: { _id: 1 },
+                     update: { '$set' => { 'x.$[i].y' => 5 } },
+                     hint: '_id_',
+                   }
+                 }]
+              end
+
+              context 'on server version 3.2 or older' do
+                max_server_fcv '3.2'
+
+                it 'raises a client error' do
+                  expect do
+                    bulk_write.execute
+                  end.to raise_error(
+                    Mongo::Error::UnsupportedCommandOption,
+                    /The hint option is only supported for bulk write with update_one commands on MongoDB server versions 3.4 and newer \(wire version 5 and greater\)/
+                  )
+                end
+              end
+            end
```

Comment:
I have only included Ruby tests for the case where the client throws an error to verify the error message. Is there an easy way for me to test that passing the hint option to the server actually worked? Is that necessary?

Reply:
To answer the first question, I believe such a thing is accomplished elsewhere in the test suite by setting up command monitoring and inspecting the command sent.